examples/tools/system/package_manager.rst

```diff
@@ -0,0 +1,229 @@
+            self.cpp_info.set_property("cmake_additional_variables_prefixes", ["CURSES",])
+
+            pkg_config = PkgConfig(self, 'ncurses')
+            pkg_config.fill_cpp_info(self.cpp_info, is_system=True)
+
+
+In this **conanfile.py** file, we are using the :ref:`system package manager<conan_tools_system_package_manager>` tool
+to install the ncurses library based on different package managers, under the
+:ref:`system_requirements<reference_conanfile_methods_system_requirements>` method. It's important to note that the
+``system_requirements`` method is called always, when building, or even if the package is already installed.
+This is useful to ensure that the package is installed in the system.
+
+Each package manager may vary the package name used to install the ncurses library, so we need to check the package manager
+documentation to find the correct package name first.
+
+Another important detail is the **package_info** method. In this method, we are using the
+:ref:`PkgConfig<conan_tools_gnu_pkgconfig>` tool to fill the **cpp_info** data, based on the file ``ncurses.pc``
+installed by the system package manager.
+
+Now, let's install the ncurses library using the **conanfile.py** file:
+
+.. code-block:: bash
+
+    $ conan create . --build=missing -c tools.system.package_manager:mode=install -c tools.system.package_manager:sudo=true
+
+Note that we are using the :ref:`Conan configuration<conan_tools_system_package_manager_config>`
+``tools.system.package_manager:mode`` as **install**, otherwise, Conan will not install the system package, but check
+if it is installed only. The same for ``tools.system.package_manager:sudo`` as **True** to run the package manager with root privileges.
+As a result of this command, you should be able to see the **ncurses** library installed in your system, in case not been installed yet.
+
+Now, let's check the **consumer** directory. This directory contains a simple C application that uses the ncurses library.
+
+The **conanfile.py** file in the **consumer** directory is:
+
+.. code-block:: python
+
+    from conan import ConanFile
+    from conan.tools.build import can_run
+    from conan.tools.cmake import cmake_layout, CMake
+    import os
+
+
+    class AppNCursesVersionConan(ConanFile):
+        settings = "os", "compiler", "build_type", "arch"
+        generators = "CMakeDeps", "CMakeToolchain"
+        package_type = "application"
+        exports_sources = "CMakeLists.txt", "ncurses_version.c"
+
+        def requirements(self):
+            if self.settings.os in ["Linux", "Macos", "FreeBSD"]:
+                self.requires("ncurses/system")
+
+        def layout(self):
+            cmake_layout(self)
+
+        def build(self):
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+
+            if self.settings.os in ["Linux", "Macos", "FreeBSD"]:
+                self.run(os.path.join(self.build_folder, "ncurses_version"), env="conanrun")
```

Comment:
Do we want to execute the example in the build method? Wouldn't it be better to execute from the executable location via the command line?  I find this a bit unexpected

Reply:
I suggested it as a way to make it simpler for both users and CI and writing the documentation. A single ``conan build .`` will work for CI and for all users, if not, we need to start dealing with:
- Different paths and layouts in Windows and Linux
- Potential activation of environment different in Windows and Linux (maybe not necessary here)

Reply:
I see your point about keeping it simple, but I think running the executable manually has more educational value and is more coherent with the rest of the docs. As a middle ground, how about printing the path to the executable at the end of `build()`, like:

```
self.output.info(f"Application built. You can run it invoking: '{os.path.join(self.build_folder, 'ncurses_version')}'")
```

Reply:
Ok, good for me then, can be done as @czoido suggests

Reply:
Done, please, see the commit: b758f39f

It also reflects the current state of the example: https://github.com/conan-io/examples2/pull/177/commits/240f393dd2fb2ac310961bdfa684b4c308909b0a